.github/workflows/eden_test.yml

```diff
@@ -1,10 +1,40 @@
 ---
-name: Test Eden WF from lf-edge/eden
-on:
-  workflow_dispatch:
+name: lf-edge/eden Test suite
+
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches:
+      - "master"
+      - "[0-9]+.[0-9]+"
+      - "[0-9]+.[0-9]+-stable"
+    paths-ignore:
+      - 'docs/**'
+      - 'pkg/pillar/docs/**'
+  pull_request_review:
+    types: [submitted]
+    paths-ignore:
+      - 'docs/**'
+      - 'pkg/pillar/docs/**'
+
+concurrency:
+  group: eden_test_suite
+  cancel-in-progress: true
 
 jobs:
-  test_suite:
+  test_suite_pr:
+    if: ${{ (github.event.review.state == 'approved') &&  ( github.event_name == 'pull_request') }}
```

Comment:
Yes, but with `concurrency` set this will cancel all in the progress workflow.

Reply:
But what if somebody will re-approve same PR with no changes? Will it cancel previous tests and run them again?

Reply:
@uncleDecart github actions are also referring the github.ref i.e the sha of the commit, IMHO if the same PR is re-approved there should be no action from github side ideally, but it would worth to test it.

Reply:
but in `concurrency` tag `github.ref` is only a fallback, which means that if pull_request_number is present (and I believe it should be present for any PR) it'll use it rather than a github-ref tag. And if we will use just github-ref in `concurrency` that will mean if somebody will force push commits or push new commits, it will be another group and old tests will be running still. Correct me if I'm wrong with `github-ref` usage 

And if GHA are smart enough not to re-trigger on re-approval I'll be happy :)